Include/httpClient/httpClient.h

```diff
@@ -3,12 +3,14 @@
 
 #pragma once
 #include "types.h"
+#include "task.h"
+#include "httpProvider.h"
+#include "mock.h"
```

Comment:
Same for this header, should only be included by people that want to do mocks, where they implement the mocks